### PR TITLE
ref(hybrid-cloud): Fixes split db testing for organization details tests

### PR DESCRIPTION
**Pull request by GabeVillalobos**
Changes:
- Updates deletion code to use region deletion schedule
- Moves delete confirmation sending code to controller, updates logic to be split-db safe
- Updates org 2fa tests to assert on a correct audit log
- Fixes audit log creation for 2fa enforcement

### CODE REVIEW

File: src/sentry/api/endpoints/organization_details.py [hunk 1]
```diff
@@ -655,6 +670,49 @@ def update_tracked_data(model):
...
+
+    organization = delete_confirmation_args.get("organization")
+    username = delete_confirmation_args.get("username")
+    user_ip_address = delete_confirmation_args.get("ip_address")
+    deletion_datetime = delete_confirmation_args.get("deletion_datetime")
+    countdown = delete_confirmation_args.get("countdown")
+
+    url = organization.absolute_url(
+        reverse("sentry-restore-organization", args=[organization.slug])
+    )
```
Comment (GabeVillalobos) [inline src/sentry/api/endpoints/organization_details.py:693]: It didn't entirely make sense to me that this lived within the organization model, so I pulled it up into the controller instead.

Comment (markstory) [reply src/sentry/api/endpoints/organization_details.py:693]: :+1: stateless module methods are always nice to see.

File: src/sentry/tasks/auth.py [hunk 2]
```diff
@@ -100,36 +104,46 @@ def remove_member(member):
...
                     ip_address=ip_address,
                     event=audit_log.get_event_id("MEMBER_PENDING"),
-                    data=member.get_audit_log_data(),
+                    data=org_member.get_audit_log_data(),
                     organization_id=org_id,
                     target_object=org_id,
                     target_user_id=user.id,
                 )
+                # Refresh the org member to ensure we always properly generate an invite link
+                org_member.refresh_from_db()
```
Comment (GabeVillalobos) [inline src/sentry/tasks/auth.py:118]: So I removed an n+1 rpc, and replaced it with an n+1 org member refresh 😅  I didn't want to change the invite link behavior to always return a link as it is used in a number of places so I kept this mostly intact.

I could duplicate this method on the rpc org member though if we really want to reduce the number of queries here.

Comment (markstory) [reply src/sentry/tasks/auth.py:118]: Database refreshing is *much* faster than an RPC across the ocean. I don't think 2fa requirements change with enough frequency that we need to sweat the additional DB queries.

**Review by markstory — APPROVED** [review]

**Pull request merged**